src/axom/core/ConduitMemory.hpp

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2017-2025, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ ******************************************************************************
+ *
+ * \file ConduitMemory.hpp
+ *
+ * \brief   Call-backs for using Axom memory mamagement in Conduit.
+ *
+ ******************************************************************************
+ */
+
+#ifndef SIDRE_CONDUITMEMORY_HPP_
+#define SIDRE_CONDUITMEMORY_HPP_
+
+// Standard C++ headers
+#include <string>
+#include <set>
+#include <memory>
+
+#include "axom/config.hpp"
+#include "axom/core/memory_management.hpp"
+#include "axom/core/utilities/Utilities.hpp"
+#include "conduit_node.hpp"
+#include "conduit_utils.hpp"
+
+namespace axom
+{
+
+/*!
+  TODO: The name isn't great.  How about ConduitMemory or ConduitAllocator MemoryForConduit?
+
+  @brief Object to handle Conduit memory operations by delegating to Axom.
+
+  This class has no public constructor.  Use instanceForAxomId(int
+  axomAllocId) to access the instance for a specific Axom allocator
+  id.  The construction registers the appropriate callbacks with
+  Conduit, including the required memset and memcopy callbacks.
+
+  Examples for setting Conduit allocator ids when you have Axom
+  allocator ids:
+
+  @code{.cpp}
+    void foo(conduit::Node& n, int axomAllocId) {
+      n.set_allocator(axomAllocIdToConduit(axomAllocId));
+    }
+
+    void bar(conduit::Node& n, int axomAllocId) {
+      const auto& instance = getInstance(axomAllocId);
+      assert(instance.axomId() == axomAllocId);
+      n.set_allocator(instance.conduitId());
+    }
+  @endcode
+*/
+struct ConduitMemory
```

Comment:
It looks like is does the same job as [ConduitAllocateThroughAxom](https://github.com/LLNL/axom/blob/7ab265aa7f3be1d3e7fc4a7f360e96ce977c8e4a/src/axom/mir/utilities/blueprint_utilities.hpp#L171]).

Reply:
It's a similar idea but it directly supports arbitrary Axom (or Umpire) allocator ids, including those of applications and third-party libraries.

Reply:
Can we extend ConduitAllocateThroughAxom to support the extended cases you need @gunney1 ?